### DISCUSSION TITLE
Transform Base Class

### DISCUSSION THREAD

**Discussion by clee-ai** [pyg-team/pytorch_geometric#2903] [Ideas]
Hi!

Right now all of the transforms are inherited from `object`, I was wondering if we could create a common ABC for all the transforms to inherit from so that I can do static typechecking. If you think that's a good idea I'd be happy to do a PR.

Thanks!

**Comment by rusty1s** [comment]
Thanks for reaching out. Can you elaborate a bit more on the possible use-cases? Right now, all transforms just implement a single `__call__(self, data)` method, [similar to how `torchvision` is doing it](https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py), so I'm not sure which benefits an abstract base class offers.

**Reply by clee-ai** [reply]
Hi, the main usecase would be for typechecking/linting so that we can validate that valid transforms are being passed as parameters or in config files:

```python
def run(transform: object):
	T.Compose([transform])
	
run("string") # typechecker won't error


def run(transform: transform_base_class): # all transforms inherit from transform_base_class
	T.Compose([transform])

run("string") # throws linter error
run(T.RandomScale()) # okay
```

Let me know if that makes sense.

**Reply by rusty1s** [reply]
I see, thanks for clarifying. I'm okay with it if you wanna contribute this feature.
